src/SFA.DAS.RoatpGateway.Web.UnitTests/Controllers/RoatpGatewayControllerTestBase.cs

```diff
@@ -37,15 +36,17 @@ protected void CoreSetup()
                 new Claim(ClaimTypes.GivenName, GivenName),
                 new Claim(ClaimTypes.Surname, Surname)
             }));
-
-            var context = new DefaultHttpContext { User = user };
+            
             GatewayValidator.Setup(v => v.Validate(It.IsAny<SubmitGatewayPageAnswerCommand>()))
                 .ReturnsAsync(new ValidationResponse
                 {
                     Errors = new List<ValidationErrorDetail>()
                 }
                 );
-            ContextAccessor.Setup(_ => _.HttpContext).Returns(context);
+
+            Context = new DefaultHttpContext { User = user };
```

Comment:
This is fine for now but won't work when you need to access the Http Request bit (for when we do file uploads).

Conveniently I wrote a nice little helper function in **_SFA.DAS.AdminService.Common.Testing.MockedObjects_** 

MockedControllerContext.Setup() will give a ControllerContext with an authenticated user and initialized Http Request

Reply:
Done